dask/array/core.py

```diff
@@ -240,6 +241,198 @@ def slices_from_chunks(chunks):
     return list(product(*slices))
 
 
+def subset_to_blocks(
+    data,
+    *,
+    coords: tuple[tuple[int]],
+    new_chunks: tuple[int] = (1, 1),
+    token="",
+):
+    token = f"{token + '-' if token else token}-subset-blocks-{tokenize(data, *coords)}"
+    keys = data._key_array[..., *coords, :]
+    graph = {}
+    for flatidx, idx in enumerate(np.ndindex(keys.shape[:-1])):
+        graph[(token, *idx[:-1], flatidx)] = tuple(keys[*idx, :])
+
+    ndim = len(coords)
+    return Array(
+        HighLevelGraph.from_collections(token, graph, dependencies=[data]),
+        token,
+        chunks=(*data._chunks[:-ndim], new_chunks),
+        meta=data,
+    )
+
+
+# Copied from Xarray
```

Comment:
what's the convention for vendoring functions?

Reply:
Not really anything, what you have now seems fine